src/pi_cam.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file
+ * @brief     Implementation for Pi camera
+ * @author    Willem de Groot, Arsalan Anwari
+ * @license   See LICENSE
+ */
+#include "pi_cam.hpp"
+
+PiCam::PiCam(const char *filePath, const Vector2D resolution, int fps)
+    : Camera(fps, resolution), filePath(std::string(filePath)), buffer(resolution.getX() * resolution.getY() * 3) {
+    correctPath();
+}
+
+void PiCam::correctPath() {
+    if (filePath.back() != '/') {
+        filePath += '/';
+    }
+    if (filePath[0] != '/') {
+        filePath.insert(0, 1, '/');
+    }
+}
+
+void PiCam::clearCommandBuffer(std::stringstream &parser) {
+    parser.str(std::string());
+    parser.clear();
+}
+
+std::string PiCam::getSettings() {
+    std::stringstream result;
+    result << " -fps " << fps << " -w " << resolution.getX() << " -h " << resolution.getY();
+    return result.str();
+}
+
+void PiCam::printSettings() {
+    std::cout << "Resolution: " << getResolution().print() << "\n"
+              << "FPS: " << getFPS() << "\n"
+              << "Store locaction: " << filePath << "\n";
+}
+
+std::string PiCam::getPath() {
+    return filePath;
+}
+
+void PiCam::setPath(const char *newPath) {
+    filePath = std::string(newPath);
+    correctPath();
+}
+
+void PiCam::takeImage(const char *name, const char *fileType) {
+    std::stringstream command;
+    command << "raspistill -o " << filePath << name << "." << fileType;
+    const char *result = command.str().c_str();
+    system(result);
+}
+
+void PiCam::takeVideo(const char *name, const unsigned int &durationMs) {
+    std::stringstream commandBuffer;
+
+    ///< Use the raspivid command to capture video with the supplied arguments
+    commandBuffer << "raspivid -o " << filePath << name << ".h264" << getSettings() << " -t " << std::to_string(durationMs);
+    const char *result1 = commandBuffer.str().c_str();

```

Comment:
```C++
const char *result1 = commandBuffer.str().c_str();
clearCommandBuffer(commandBuffer);
system(result1);
```
Bit of a weird construction especially with result1, result2 and result3. I'm advising to reorder these commands like this:
```C++
system(commandBuffer.str().c_str());
clearCommandBuffer(commandBuffer);
```